### ISSUE TITLE
Installation failure in terminal on some machines

### ISSUE THREAD

**Issue by fontikar** [traitecoevo/austraits#46]
**Describe the bug**
Installation failure in terminal on some machines (@dfalster's terminal), seems to be an issue with pandoc as a key dependency to build vignettes

**To Reproduce**
#install.packages("remotes")
remotes::install_github("traitecoevo/austraits", build_vignettes = TRUE)

**Expected behavior**
A successful build and calling vignette("austraits") will be error free

**Error messages**
 Error: Vignette re-building failed.
   Execution halted
Error: Failed to install 'austraits' from GitHub:
  System command 'R' failed, exit status: 1, stdout + stderr (last 10 lines):
E> --- re-building 'structure.Rmd' using rmarkdown
E> Error: processing vignette 'structure.Rmd' failed with diagnostics:
E> Pandoc is required to build R Markdown vignettes but not available. Please make sure it is installed.
E> --- failed re-building 'structure.Rmd'
Hmms, seems Rstudio can find pandoc but R in terminal can't
#Rstudio
> rmarkdown::pandoc_version()
[1] '2.14.0.3'

# Terminal

> rmarkdown::pandoc_version()
[1] '0'

**Comment by fontikar** [comment]
Seems to be happening with other users in RStudio

![image001](https://user-images.githubusercontent.com/12161802/168700419-74c8b581-bfdc-4d2c-836e-d4962b49c3cd.png)


But it looks like this user (Payal) didn't have some dependencies on their machine (they recently updated R which is already a bit pain haha), so I suggested: 
`remotes::install_github("traitecoevo/austraits", dependencies = TRUE, upgrade = "ask", build_vignettes = TRUE)` 
so they can install all the dependencies and upgrade any that are out of date. 

Lets see how that goes

**Comment by jrfep** [comment]
I also failed to install in katana on demand, I got a similar error about the vignette `structure.Rmd` and the package `kableExtra`.
I opted to install the package with `build_vignettees=FALSE` for now, let's see how it goes.

**Comment by tinaparky** [comment]
@fontikar I am also unable to build the package on RStudio, using either install code (with or without dependencies = TRUE). 
E  creating vignettes (5.2s)
   --- re-building ‘austraits.Rmd’ using rmarkdown
   Error: processing vignette 'austraits.Rmd' failed with diagnostics:
   object ‘faststack’ is not exported by 'namespace:fastmap'
   --- failed re-building ‘austraits.Rmd’
   
   --- re-building ‘dictionary.Rmd’ using rmarkdown
   Error: processing vignette 'dictionary.Rmd' failed with diagnostics:
   object ‘faststack’ is not exported by 'namespace:fastmap'
   --- failed re-building ‘dictionary.Rmd’
   
   --- re-building ‘structure.Rmd’ using rmarkdown
   Error: processing vignette 'structure.Rmd' failed with diagnostics:
   object ‘faststack’ is not exported by 'namespace:fastmap'
   --- failed re-building ‘structure.Rmd’
   
   SUMMARY: processing the following files failed:
     ‘austraits.Rmd’ ‘dictionary.Rmd’ ‘structure.Rmd’
   
   Error: Vignette re-building failed.
   Execution halted
Error: Failed to install 'austraits' from GitHub:
  System command 'R' failed, exit status: 1, stdout + stderr (last 10 lines):
E> --- re-building ‘structure.Rmd’ using rmarkdown
E> Error: processing vignette 'structure.Rmd' failed with diagnostics:
E> object ‘faststack’ is not exported by 'namespace:fastmap'
E> --- failed re-building ‘structure.Rmd’
E> 
E> SUMMARY: processing the following files failed:
E>   ‘austraits.Rmd’ ‘dictionary.Rmd’ ‘structure.Rmd’
E> 
E> Error: Vignette re-building failed.
E> Execution halted

with dependencies: 
  --- re-building ‘austraits.Rmd’ using rmarkdown
   Quitting from lines 17-30 (austraits.Rmd) 
   Error: processing vignette 'austraits.Rmd' failed with diagnostics:
   package or namespace load failed for 'dplyr' in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]):
    namespace 'vctrs' 0.3.4 is already loaded, but >= 0.3.8 is required
   --- failed re-building ‘austraits.Rmd’
   
   --- re-building ‘dictionary.Rmd’ using rmarkdown
   Error: processing vignette 'dictionary.Rmd' failed with diagnostics:
   'set_envvar' is not an exported object from 'namespace:xfun'
   --- failed re-building ‘dictionary.Rmd’
   
   --- re-building ‘structure.Rmd’ using rmarkdown
   Quitting from lines 12-37 (structure.Rmd) 
   Error: processing vignette 'structure.Rmd' failed with diagnostics:
   package or namespace load failed for 'dplyr' in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]):
    namespace 'vctrs' 0.3.4 is already loaded, but >= 0.3.8 is required
   --- failed re-building ‘structure.Rmd’
   
   SUMMARY: processing the following files failed:
     ‘austraits.Rmd’ ‘dictionary.Rmd’ ‘structure.Rmd’
   
   Error: Vignette re-building failed.
   Execution halted
Error: Failed to install 'austraits' from GitHub:
  System command 'R' failed, exit status: 1, stdout + stderr (last 10 lines):
E> Error: processing vignette 'structure.Rmd' failed with diagnostics:
E> package or namespace load failed for 'dplyr' in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]):
E>  namespace 'vctrs' 0.3.4 is already loaded, but >= 0.3.8 is required
E> --- failed re-building ‘structure.Rmd’
E> 
E> SUMMARY: processing the following files failed:
E>   ‘austraits.Rmd’ ‘dictionary.Rmd’ ‘structure.Rmd’
E> 
E> Error: Vignette re-building failed.
E> Execution halted
In addition: There were 12 warnings (use warnings() to see them)
> 

Any ideas what could be causing this?

**Comment by fontikar** [comment]
Hi @tinaparky apologies for a late reply! Lets get you up and running ASAP!
I suspect there are two problems at hand:
1.  various packages on your computer that are out of date
2. some issue with building of vignettes (we've bee having heaps of trouble with this since the beginning!) 

I recommend you for now, skip the building of vignettes and try update all the packages on your computer. This line of code should install the dependencies needed for `austraits` and upgrading any exisiting ones that is needed by the package. You will need to follow the prompts in the R console! i.e. selecting `1 = All` when asked if you want to update your packages.
`remotes::install_github("traitecoevo/austraits", dependencies = TRUE, upgrade = "ask")`

Let me know how that works for you!

**Comment by fontikar** [comment]
Related to #47

**Comment by fontikar** [comment]
In the latest patch release, we've precompiled all vignettes so they don't require on the fly building using dependencies. This should streamline the installation process and avoid hiccups addressed in this issue. We've also included more detailed instructions in the README for installation, by setting `dependencies = TRUE` and `upgrade = ask` it forces the user install all dependencies (Imports, depends, suggests) to actively update any dependencies they have already installed.
